modelzoo/DIN/train.py

```diff
@@ -1,387 +1,739 @@
-                        help='set the number of steps on test training model',
-                        type=int,
-                        default=100)
-    parser.add_argument('--steps',
-                        help='set the number of steps on train dataset',
-                        type=int,
-                        default=0)
-    parser.add_argument('--timeline',
-                        help='number of steps on saving timeline. Default 0',
+    parser.add_argument('--keep_checkpoint_max',
+                        help='Maximum number of recent checkpoint to keep',
                         type=int,
-                        default=0)
+                        default=1)
     parser.add_argument('--bf16',
                         help='enable DeepRec BF16 in deep model. Default FP32',
                         action='store_true')
     parser.add_argument('--no_eval',
                         help='not evaluate trained model by eval dataset.',
                         action='store_true')
+    parser.add_argument('--timeline',
+                        help='number of steps on saving timeline. Default 0',
+                        type=int,
+                        default=0)
+    parser.add_argument("--protocol",
+                        type=str,
+                        choices=["grpc", "grpc++", "star_server"],
+                        default="grpc")
+    parser.add_argument('--inter',
+                        help='set inter op parallelism threads.',
+                        type=int,
+                        default=0)
+    parser.add_argument('--intra',
+                        help='set inter op parallelism threads.',
+                        type=int,
+                        default=0)
+    parser.add_argument('--input_layer_partitioner',
+                        help='slice size of input layer partitioner. units MB',
+                        type=int,
+                        default=0)
+    parser.add_argument('--dense_layer_partitioner',
+                        help='slice size of dense layer partitioner. units KB',
+                        type=int,
+                        default=0)
     return parser
 
 
-if __name__ == '__main__':
+def main(tf_config=None, server=None):
+    # check dataset
+    print('Checking dataset')
+    train_file = args.data_location + '/local_train_splitByUser'
+    test_file = args.data_location + '/local_test_splitByUser'
+
+    if (not os.path.exists(train_file)) or (not os.path.exists(test_file)):
+        print("Dataset does not exist in the given data_location.")
+        sys.exit()
+    no_of_training_examples = sum(1 for line in open(train_file))
+    no_of_test_examples = sum(1 for line in open(test_file))
+    print("Numbers of training dataset is {}".format(no_of_training_examples))
+    print("Numbers of test dataset is {}".format(no_of_test_examples))
+
+    # set params
+    # set batch size & steps
+    batch_size = args.batch_size
+    if args.steps == 0:
+        no_of_epochs = 3
```

Comment:
Why is number of epochs hard-coded to 3 in the script for args.steps equal 0?

Reply:
I just use `args.steps`  when I want to test the model, get the model's timeline, and benchmark gsteps. When tuning ACC/AUC, I can just use the command `python train.py` for training a specific number of epochs.

Reply:
Fair enough